gtime/plotting/plotting.py

```diff
@@ -0,0 +1,267 @@
+import matplotlib.pyplot as plt
+import pandas as pd
+import numpy as np
+from gtime.plotting.preprocessing import seasonal_split, acf, pacf
+from scipy.stats import norm
+
+
+def lag_plot(df: pd.DataFrame, lags, plots_per_row: int = 4):
+    """
+    Lag plots, scatter plot of x_i against x_i-k for every k in ``lags``.
+    https://www.itl.nist.gov/div898/handbook/eda/section3/lagplot.htm
+
+    Parameters
+    ----------
+    df : pd.DataFrame, time series to plot
+    lags : int or list of ints, lags to plot
+    plots_per_row : int, number of lag plots per one row of seasonal_subplots
+
+    Returns
+    -------
+    axes : plot axes
+
+    Examples
+    --------
+    >>> import pandas as pd
+    >>> import numpy as np
+    >>> import matplotlib.pyplot as plt
+    >>> from gtime.plotting import lag_plot
+    >>> from gtime.forecasting import NaiveForecaster
+    >>> idx = pd.period_range(start='2011-01-01', end='2012-01-01')
+    >>> np.random.seed(1)
+    >>> df = pd.DataFrame(np.random.random((len(idx), 1)), index=idx, columns=['1'])
+    >>> lag_plot(df, lags=[1, 5, 10])
+    >>> plt.show()
+
+    """
+
+    if isinstance(lags, int):
+        lags = list(range(1, lags + 1))
+
+    if len(lags) > plots_per_row:
+        rows = (len(lags) - 1) // plots_per_row + 1
+        cols = plots_per_row
+    else:
+        rows = 1
+        cols = len(lags)
+
+    fig, ax = plt.subplots(
+        rows,
+        cols,
+        sharey=True,
+        sharex=True,
+        figsize=(20, 5 * rows),
+        squeeze=False,
+        gridspec_kw={"wspace": 0.05},
+    )
+    x_lim = df.agg(["min", "max"]).values
+
+    for i, l in enumerate(lags):
+        axes = ax[i // plots_per_row, i % plots_per_row]
+        axes.scatter(df.iloc[l:], df.iloc[:-l])
+        axes.set(title="Lag " + str(l))
+        axes.plot(x_lim, x_lim, ls="--", c=".7")
+        axes.set(xlim=x_lim, ylim=x_lim)
+        axes.label_outer()
+
+    return ax
+
+
+def seasonal_subplots(df: pd.DataFrame, cycle, freq=None, agg="mean", box=False):
```

Comment:
Maybe calling it `seasonal_subseries_plot` is more appropriate? What do you think?

Reply:
Also, add typing for `cycle`

Reply:
In my opinion such long names can be inconvenient if the function is expected to be often called by users. What do you think?
As for typing - added for all arguments